### DISCUSSION TITLE
Reading/Writing Struct into TOML

### DISCUSSION THREAD

**Discussion by ghost** [toml-lang/toml#831] [General]
Hi I have a large SoilWater-ToolBox model and I have got all the input parameters, Parameters, as a mutable struct. In order to create a package I will need to read the parameters outside the model as recommended in previous post as a TOML format.

**So the question is how to convert the structure into TOML?
Then how to read TOML file into a structure?**

I will be delighted if one can provide me guidance how to perform this task for example by using the simple structure:

```
mutable struct OPTION
   transpiration:: TRANSPIRATION
   soil:: SOIL 
end

mutable struct TRANSPIRATION
      Evaporation :: Bool
      Transpiration :: Bool
end

mutable struct SOIL
      Topsoil :: Bool
      Macropore :: Bool
end

# ====
Evaporation = true
Transpiration = false
transpiration = TRANSPIRATION(Evaporation, Transpiration)

Topsoil = true
Macropore = false
soil = SOIL(Topsoil, Macropore)

option = OPTION(transpiration, soil)

```
Many thanks for any help you may provide,
Joseph

**Comment by arp242** [comment]
What language/environment are you using?

You have to use an implementation for your environment; there's a list here: https://github.com/toml-lang/toml/wiki#implementations

**Comment by ghost** [comment]
Dear Martin,

A great thanks I am using Julia language.
And thanks for your website.

Cheers,
Joseph

**Comment by arp242** [comment]
Cheers, https://github.com/JuliaLang/TOML.jl is listed, but it seems it's in the standard library:

> Note: In Julia 1.6+, this is a standard library and does not need to be explicitly installed.

This is the general discussion for the TOML file format; I never programmed Julia so I can't really help you beyond that. If you have problems with it, then you're probably better off asking this on a more Julia-specific location as you're much more likely to get a useful answer.

**Comment by ghost** [comment]
Dear martin,

A great thanks for your useful information. I will contact JULIA user group to answer the specific question.

Thanks for your help,
Joseph
